python/plugins/processing/tools/postgis.py

```diff
@@ -59,6 +59,9 @@ def uri_from_name(conn_name):
 
     settings.endGroup()
 
+    if hasattr(authcfg, 'isNull') and authcfg.isNull():
```

Comment:
This part looks a bit awkward to me
Why is the hasattr(authcfg, 'isNull') needed ? Are there case where authcfg wouldn't be a QString ?
And then why transforming a null QString into an empty QString ?
 Shouldn't rather setConnection() be made robust to support null strings in addition to empty strings.